src/middlewared/middlewared/main.py

```diff
@@ -1753,8 +1753,10 @@ async def __initialize(self):
         if await self.call('system.state') == 'READY':
             self._setup_periodic_tasks()
 
+        unix_socket_path = os.path.join(MIDDLEWARE_RUN_DIR, 'middlewared.sock')
         await web.TCPSite(runner, '0.0.0.0', 6000, reuse_address=True, reuse_port=True).start()
-        await web.UnixSite(runner, os.path.join(MIDDLEWARE_RUN_DIR, 'middlewared.sock')).start()
+        await web.UnixSite(runner, unix_socket_path).start()
+        os.chmod(unix_socket_path, 0o666)
```

Comment:
Just another small comment while I'm here. `os.chmod` is blocking but being called in main event loop.

Reply:
We do a lot of blocking I/O at this point (e.g. importing plugin modules). There is no easy fix for this (middleware and threading pool are inaccessible here), but I also see no issue with performing blocking I/O in initialization since we are not doing anything concurrently before we start accepting requests (which happens much later). What do you think?

Reply:
Fair point.